Java/src/com/gymfox/httpserver/HTTPServer.java

```diff
@@ -0,0 +1,115 @@
+package com.gymfox.httpserver;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static com.gymfox.httpserver.HTTPServerExceptions.HttpServerIsRunningException;
+import static com.gymfox.httpserver.HTTPServerExceptions.InvalidRequestParametersCountException;
+import static com.gymfox.httpserver.HTTPServerUtils.*;
+
+public class HTTPServer {
+    private final ExecutorService pool;
+    private final HTTPServerConf httpServerConf;
+    static HTTPMimeTypes mimeTypeFile;
+
+    private volatile ServerSocket serverSocket;
+    private volatile boolean isRunning;
+
+    public HTTPServer() throws IOException {
+        this(CONFIG_FILE);
+    }
+
+    public HTTPServer(File pathToConfigFile) throws IOException {
+        httpServerConf = ConfigSerializer.getHTTPConfig(pathToConfigFile);
+        mimeTypeFile = ConfigSerializer.getMimeTypes(httpServerConf.getConfigMimeTypes());
+        pool = Executors.newFixedThreadPool(httpServerConf.getPoolSize());
+    }
+
+    public void start() throws IOException {
+        runHttpServer();
+
+        while (isRunning()) {
+            Socket socket = serverSocket.accept();
+            pool.execute(() -> {
+                try (PrintWriter sout = new PrintWriter(new OutputStreamWriter(socket.getOutputStream()));
+                     BufferedReader sin = new BufferedReader(new InputStreamReader(socket.getInputStream()))) {
+                    System.out.println("Client has been connected");
+
+                    if (isRunning()) {
+                        HTTPTransformer httpTransformer = new HTTPTransformer();
```

Comment:
Can make as a field as it doesn't have any resources specific for each request.
So a single transformer can be used for all requests